li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/LiKafkaFederatedConsumerImpl.java

```diff
@@ -214,8 +214,7 @@ synchronized public void subscribe(Collection<String> topics, ConsumerRebalanceL
     throw new UnsupportedOperationException("Not implemented yet");
   }
 
-  @Override
-  synchronized public void assign(Collection<TopicPartition> partitions) {
+  private Map<ClusterDescriptor, Set<TopicPartition>> getPerClusterTopicPartitionSet(Collection<TopicPartition> partitions) {
     ensureOpen();
```

Comment:
It appears that this is supposed to be a generic utility, but it contains assign-specific code here and there (lines 218-228, 252). 

I also created a similar utility in my pr (https://github.com/linkedin/li-apache-kafka-clients/pull/119) - can you copy getPartitionsByCluster() (https://github.com/linkedin/li-apache-kafka-clients/pull/119/files#diff-5b1f24259bdffe649d28e50386b8a3faR646) and use it here to reduce merge conflicts?

Reply:
Sure.